cmd/gnodev/precompile.go

```diff
@@ -58,7 +64,18 @@ func precompileApp(cmd *command.Command, args []string, iopts interface{}) error
 	return nil
 }
 
-func precompilePkg(pkgPath string, opts precompileOptions) error {
+func precompilePkg(pkgPath string, opts *precompileOptions) error {
+	if opts.Output != defaultPrecompileOptions.Output {
+		if err := os.MkdirAll(filepath.Join(opts.Output, pkgPath), 0o755); err != nil {
+			return err
+		}
+	}
+
+	if opts.skipPkgs[pkgPath] {
+		return nil
+	}
+	opts.skipPkgs[pkgPath] = true
```

Comment:
Please see my previous comment about dropping `skipPkgs` from the options - you can simply use a map at the top level (where the loop is being called) and keep a tally of what packages you need to run `precompilePkg` against - this way `precompilePkg` doesn't have a side effect of skipping over files based on modifiable options